test/suite/test_hs20.py

```diff
@@ -60,46 +60,38 @@ def test_hs20(self):
         value1 = 'a' * 500
         value2 = 'b' * 50
 
-        # FIXME-WT-9063 revisit the use of self.retry() throughout this file.
-
         # Insert a value that is larger than the maximum leaf value.
         for i in range(0, 10):
-            for retry in self.retry():
-                with retry.transaction(commit_timestamp = 2):
-                    cursor[self.make_key(i)] = value1
+            with self.transaction(commit_timestamp = 2):
+                cursor[self.make_key(i)] = value1
 
         # Do 2 modifies.
         for i in range(0, 10):
-            for retry in self.retry():
-                with retry.transaction(commit_timestamp = 3):
-                    cursor.set_key(self.make_key(i))
-                    mods = [wiredtiger.Modify('B', 500, 1)]
-                    self.assertEqual(cursor.modify(mods), 0)
+            with self.transaction(commit_timestamp = 3):
+                cursor.set_key(self.make_key(i))
+                mods = [wiredtiger.Modify('B', 500, 1)]
+                self.assertEqual(cursor.modify(mods), 0)
 
         for i in range(0, 10):
-            for retry in self.retry():
-                with retry.transaction(commit_timestamp = 4):
-                    cursor.set_key(self.make_key(i))
-                    mods = [wiredtiger.Modify('C', 501, 1)]
-                    self.assertEqual(cursor.modify(mods), 0)
+            with self.transaction(commit_timestamp = 4):
+                cursor.set_key(self.make_key(i))
+                mods = [wiredtiger.Modify('C', 501, 1)]
+                self.assertEqual(cursor.modify(mods), 0)
 
         # Insert more data to trigger eviction.
         for i in range(10, 100000):
-            for retry in self.retry():
-                with retry.transaction(commit_timestamp = 5):
-                    cursor[self.make_key(i)] = value2
+            with self.transaction(commit_timestamp = 5):
+                cursor[self.make_key(i)] = value2
 
         # Update the overflow values.
         for i in range(0, 10):
-            for retry in self.retry():
-                with retry.transaction(commit_timestamp = 5):
-                    cursor[self.make_key(i)] = value2
+            with self.transaction(commit_timestamp = 5):
+                cursor[self.make_key(i)] = value2
 
         # Do a checkpoint to move the overflow values to the history store but keep the current in memory disk image.
         self.session.checkpoint()
 
         # Search the first modifies.
         for i in range(0, 10):
-            for retry in self.retry():
-                with retry.transaction(read_timestamp = 3, rollback = True):
-                    self.assertEqual(cursor[self.make_key(i)], value1 + "B")
+            with self.transaction(read_timestamp = 3, rollback = True):
```

Comment:
Do all tests that use explicit transactions need this now (even if they don't specify a timestamp arg)?

Reply:
No, it's entirely optional to use self.transaction().  Since self.transaction() doesn't do retries anymore, the original code in test_hs20.py and the PR code are equivalent.